libhoney.go

```diff
@@ -334,10 +343,18 @@ func Init(config Config) error {
 	if config.PendingWorkCapacity == 0 {
 		config.PendingWorkCapacity = DefaultPendingWorkCapacity
 	}
+	if config.Logger == nil {
+		config.Logger = &nullLogger{}
```

Comment:
If the user doesn't ever call `libhoney.Init()`, e.g. in unit tests, then `libhoney` will panic on calls to `logger.Log`. I think we should try to avoid this. Either the global `logger` should be initialized to a `nullLogger`, or we should define something like

```
func log(msg string, args ...interface{}) {
  if logger == nil {
    return
  }
  // ...
}
```

and use that in place of `logger.Log`.

Reply:
how about setting `logger Logger = &nullLogger{}` as the global instead of leaving it nil?